controlplane/kubeadm/controllers/controller.go

```diff
@@ -427,28 +427,30 @@ func (r *KubeadmControlPlaneReconciler) reconcileHealth(ctx context.Context, clu
 
 	// Do a health check of the Control Plane components
 	if err := r.managementCluster.TargetClusterControlPlaneIsHealthy(ctx, util.ObjectKey(cluster)); err != nil {
-		logger.V(2).Info("Waiting for control plane to pass control plane health check to continue reconciliation", "cause", err)
+		logger.V(2).Info("Waiting for control plane to pass control plane health check to continue reconciliation", "cause", err.Error())
 		r.recorder.Eventf(kcp, corev1.EventTypeWarning, "ControlPlaneUnhealthy",
 			"Waiting for control plane to pass control plane health check to continue reconciliation: %v", err)
 		return &capierrors.RequeueAfterError{RequeueAfter: healthCheckFailedRequeueAfter}
 	}
 
-	// Ensure etcd is healthy
-	if err := r.managementCluster.TargetClusterEtcdIsHealthy(ctx, util.ObjectKey(cluster)); err != nil {
-		// If there are any etcd members that do not have corresponding nodes, remove them from etcd and from the kubeadm configmap.
-		// This will solve issues related to manual control-plane machine deletion.
-		workloadCluster, err := r.managementCluster.GetWorkloadCluster(ctx, util.ObjectKey(cluster))
-		if err != nil {
-			return err
-		}
-		if err := workloadCluster.ReconcileEtcdMembers(ctx); err != nil {
-			logger.V(2).Info("Failed attempt to remove potential hanging etcd members to pass etcd health check to continue reconciliation", "cause", err)
-		}
+	// If KCP should manage etcd, ensure etcd is healthy.
+	if controlPlane.ShouldManageEtcd() {
```

Comment:
Can we flip this? Managed etcd is the 80% use case, and there is going to be more code to manage etcd in the if blocks than not. If we do `controlPlane.IsEtcdExternal`, we could just return.

Reply:
Actually nevermind, seems it's not really worth it. I'd rename it to `IsEtcdManaged` though